### PR TITLE
[SC7201] Spike POC for the option pricer model using quantlib

### PR DESCRIPTION
**Pull request by AnilSorathiya**
## Internal Notes for Reviewers
This changes creates a demo notebook for the Heston option price using an external library:
* Use of QuantLib library to build a model
* Data sourced dynamically from yahoo finance
* Model parameter calculated based on under line stock data
* Add custom tests for benchmarking, sensitivity and stress testing

## External Release Notes
This changes creates a demo notebook for the Heston option price using an external library:
* Use of QuantLib library to build a model
* Data sourced dynamically from yahoo finance
* Model parameter calculated based on under line stock data
* Add custom tests for benchmarking, sensitivity and stress testing

### CODE REVIEW

File: notebooks/code_sharing/capital_markets/quickstart_option_pricing_models_quantlib.ipynb [hunk 1]
```diff
@@ -0,0 +1,1088 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from scipy.optimize import minimize\n",
+    "import yfinan\n",
+    "importib as ql\n",
```
Comment (cachafla) [inline notebooks/code_sharing/capital_markets/quickstart_option_pricing_models_quantlib.ipynb:253]: Typo here 😬 

```
import quantlib as ql
```

Comment (cachafla) [reply notebooks/code_sharing/capital_markets/quickstart_option_pricing_models_quantlib.ipynb:253]: Actually, still get an error:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[5], [line 7](vscode-notebook-cell:?execution_count=5&line=7)
      [5](vscode-notebook-cell:?execution_count=5&line=5) from scipy.optimize import minimize
      [6](vscode-notebook-cell:?execution_count=5&line=6) import yfinance as yf
----> [7](vscode-notebook-cell:?execution_count=5&line=7) import quantlib as ql
      [8](vscode-notebook-cell:?execution_count=5&line=8) from validmind.tests import run_test

ModuleNotFoundError: No module named 'quantlib'
```

Comment (AnilSorathiya) [reply notebooks/code_sharing/capital_markets/quickstart_option_pricing_models_quantlib.ipynb]: it's  `QuantLib`. Updated the notebook now.

**Review by cachafla — APPROVED** [review]
Looks great 👏 .

Before merging I recommend adjusting the text a little bit on some sections for consistency. Some of them say "let's evaluates" and some others say "XYZ test evaluates"

### Example: before

```
Let's evaluates the sensitivity of a model's output to changes in the kappa parameter, which is a mean reversion rate in stochastic volatility models.
```

### Example: after

```
This test evaluates the sensitivity of a model's output to changes in the kappa parameter, which is a mean reversion rate in stochastic volatility models.
```

Also, some sections have text that seems to be coming from somewhere else? Example:

### Model Calibration
* Clearly state the purpose of the calibration process. For example, in the context of an option pricing model, calibration aims to adjust model parameters to fit market data (e.g., market option prices, volatility surfaces).
* Specify whether the calibration is to historical data, current market data, or a blend of both.

"Clearly state" sounds confusing in this context. Maybe it could be something like:

### Model Calibration
* The calibration process aims to optimize the Heston model parameters (v0, theta, kappa, sigma, rho) by minimizing the difference between model-predicted option prices and observed market prices.
* In this implementation, the model is calibrated to current market data, specifically using option prices from the selected ticker and expiration date.

Anyway. I'd suggest some similar adjustments in the text before merging.

**Pull request merged**